.github/workflows/main.yml

```
@@ -0,0 +1,52 @@
+name: CI
+on: 
+  pull_request:
+  push:
+    branches:
+      - main
+      - add_GHA_based_CI
```

```suggestion
workflow_dispatch:
```

I understand why you had this before when you were adding the workflow to get things running, but as you're targeting `main` in this PR you can remove this before merge.

You can instead add a `workflow_dispatch` "`on`" trigger which allows you to run on demand from GitHub's web interface.